modules/text2vec-google/vectorizer/objects_texts.go

```diff
@@ -61,25 +52,28 @@ func (v *Vectorizer) Object(ctx context.Context, object *models.Object, cfg modu
 func (v *Vectorizer) object(ctx context.Context, object *models.Object, cfg moduletools.ClassConfig,
 ) ([]float32, error) {
 	icheck := NewClassSettings(cfg)
-
 	corpi, titlePropertyValue := v.objectVectorizer.TextsWithTitleProperty(ctx, object, icheck, icheck.TitleProperty())
 	// vectorize text
-	res, err := v.client.Vectorize(ctx, []string{corpi}, ent.VectorizationConfig{
-		ApiEndpoint: icheck.ApiEndpoint(),
-		ProjectID:   icheck.ProjectID(),
-		Model:       icheck.Model(),
-		Dimensions:  icheck.Dimensions(),
-		TaskType:    icheck.TaskType(),
-	}, titlePropertyValue)
+	res, err := v.client.VectorizeWithTitleProperty(ctx, []string{corpi}, titlePropertyValue, cfg)
 	if err != nil {
 		return nil, err
 	}
-	if len(res.Vectors) == 0 {
+	if len(res.Vector) == 0 {
 		return nil, fmt.Errorf("no vectors generated")
 	}
 
-	if len(res.Vectors) > 1 {
-		return libvectorizer.CombineVectors(res.Vectors), nil
+	if len(res.Vector) > 1 {
+		return libvectorizer.CombineVectors(res.Vector), nil
+	}
+	return res.Vector[0], nil
+}
+
+func (v *Vectorizer) Texts(ctx context.Context, inputs []string,
+	cfg moduletools.ClassConfig,
+) ([]float32, error) {
+	res, err := v.client.VectorizeQuery(ctx, inputs, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("remote client vectorize: %w", err)
 	}
-	return res.Vectors[0], nil
+	return libvectorizer.CombineVectors(res.Vector), nil
 }
```

Comment:
I think this should be all shared with the other modules and can be deleted here